src/recipe_dictionary.cpp

```diff
@@ -33,9 +33,57 @@ std::vector<const recipe *> recipe_subset::search( const std::string &txt ) cons
 {
     std::vector<const recipe *> res;
 
+    size_t split;
+    split = txt.find( ":" );
+    std::string key = "", value;
+    if( split != std::string::npos ) {
+        key = txt.substr( 0, split );
+        value = txt.substr( split + 1 );
+    } else {
+        value = txt;
+    }
+
     std::copy_if( recipes.begin(), recipes.end(), std::back_inserter( res ),
-    [ &txt ]( const recipe * r ) {
-        return lcmatch( item::nname( r->result ), txt );
+    [&]( const recipe * r ) {//sadly it seems easier to copy the scope here for the moment
+        if( key == _( "Q" ) ) { // result quality match
+            const auto &result = item( r->result );
```

Comment:
Constructing an `item` just to access `type` is very inefficient. You can use `item::find_type` which would be hugely more efficient